examples/grpc/src/main/java/io/opentelemetry/example/HelloWorldServer.java

```diff
@@ -0,0 +1,184 @@
+          if (carrier.containsKey(k)) {
+            return carrier.get(k);
+          }
+          return "";
+        }
+      };
+
+  public HelloWorldServer() {
+    initTracer();
+  }
+
+  private void start() throws IOException {
+    /* The port on which the server should run */
+
+    server =
+        ServerBuilder.forPort(this.port)
+            .addService(new GreeterImpl())
+            // Intercept gRPC calls
+            .intercept(new OpenTelemetryServerInterceptor())
+            .build()
+            .start();
+    logger.info("Server started, listening on " + port);
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread() {
+              @Override
+              public void run() {
+                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+                System.err.println("*** shutting down gRPC server since JVM is shutting down");
+                HelloWorldServer.this.stop();
+                System.err.println("*** server shut down");
+                System.err.println(
+                    "*** forcing also the Tracer Exporter to shutdown and process the remaining traces");
+                inMemexporter.shutdown();
+                System.err.println("*** Trace Exporter shut down");
+              }
+            });
+  }
+
+  private void stop() {
+    if (server != null) {
+      server.shutdown();
+    }
+  }
+
+  private void initTracer() {
+    // Get the tracer
+    TracerSdkFactory tracerFactory = OpenTelemetrySdk.getTracerFactory();
+    // Set to process in memory the spans
+    tracerFactory.addSpanProcessor(SimpleSpansProcessor.newBuilder(inMemexporter).build());
+    // Give the name to the traces
+    this.tracer = tracerFactory.get("io.opentelemetry.example.HelloWorldServer");
+    textFormat = this.tracer.getHttpTextFormat();
+  }
+
+  /** Await termination on the main thread since the grpc library uses daemon threads. */
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+  class GreeterImpl extends GreeterGrpc.GreeterImplBase {
+
+    @Override
+    public void sayHello(HelloRequest req, StreamObserver<HelloReply> responseObserver) {
+      // Extract the Context from the gRPC request
+      SpanContext ctx = TRACE_ID_CTX_KEY.get();
+      // Build a span based on the received context
+      Span span =
+          tracer
+              .spanBuilder("helloworld.Greeter/SayHello")
+              .setParent(ctx)
+              .setSpanKind(Span.Kind.SERVER)
+              .startSpan();
+      // TODO provide attributes to Span.Builder
+      span.setAttribute("component", "grpc");
+      span.setAttribute("rpc.service", "helloworld.Greeter");
+      span.setAttribute("net.peer.name", "localhost");
+      span.setAttribute("net.peer.port", HelloWorldServer.this.port);
```

Comment:
`net.peer` describes the remote endpoint, in this case the client.

You could set the server's port on `net.host.port`, however.

Reply:
The following two specifications are misleading: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-span-general.md#net.name

Reply:
You're right. I opened https://github.com/open-telemetry/opentelemetry-specification/pull/386.